### ISSUE TITLE
Improve update frequency for reserved sequences with old progress

### ISSUE THREAD

**Issue by dubslow** [MersenneForum/MersenneForumAliquot#12]
Per 5bcc009c8, I don't really want to special case yafu@home, if the current system isn't good enough for yafu@home then we should fix the system rather than special case hack it :)

**Comment by ChristianBeer** [comment]
yafu@home is a bit different from the over reservation system users. Some have sequences reserved with a high cofactor where the sequence was not advanced within 2 years. It is very unlikely that they will report progress where on the other hand yafu delivers progress within days if it has a reservation on a seq.

The other idea I had is to alter the priority update algorithm to take cofactor size and reservation status in mind and lower the priority even more if reserved and small cofactor because a sequence is more likely to advance (when run on yafu@home) if there is a small cofactor. I couldn't figure out how to translate that into a math formula to feed to update algorithm and it would still include a special case. Maybe we can eliminate the special yafu@home case by lowering priority on general for all mass_reservers since they tend to be faster than usual forum people?

**Comment by dubslow** [comment]
There already is a conditional to reduce priority for small cofactors: https://github.com/MersenneForum/MersenneForumAliquot/blob/master/mfaliquot/application/__init__.py#L165

I had it even stronger before, but then there were small cofactors that were unreserved that were getting queried a whole bunch for nothing.

Maybe I should strengthen it again, since on MersenneForum we know there is at least one user running spiders to factor all FDB composites < 98 digits.

The more fundamental problem, as I've come to see it in the last couple hours, is that the underlying problem we're discussing here is that we need a way to increase priority for sequences whose `date of reservation` is *after* their `date of last progress`. This is the most general description; I suspect most of the sequences found by that script of yours match this description. And, it applies not only to yafu@home but also to all the usual manual reservations as well, while it does not apply to those very large sequences which don't advance for years at a time because they have made progress since acquiring their reservation.

Of course we can't directly prioritize those sequences since we don't store the date-of-reservation, and the last thing this repository needs is Even More Statefulness!™ :)

Perhaps a decent compromise "solution", pseudo-hack, would be to make it so that all reserved sequences have a maximum update period of 30 days instead of the non-reserved max period of 90 days? Perhaps that and *not* update newly-reserved sequences, as is done currently but possibly counterproductively?

**Comment by ChristianBeer** [comment]
> Perhaps a decent compromise "solution", pseudo-hack, would be to make it so that all reserved sequences have a maximum update period of 30 days instead of the non-reserved max period of 90 days? Perhaps that and not update newly-reserved sequences, as is done currently but possibly counterproductively?

I agree with both. Lowering the max update time for reserved sequences would make the yafu situation better. But maybe instead of not updating newly reserved sequences we can lower the priority to such a level that it will be updated within the next 7 days? That would allow the reservee to update the seq in FDB and then the normal priority based update can follow.

**Comment by dubslow** [comment]
The problem with priority adjustments is that they would get overridden every 6 hours by the script. There's currently no easy to mark a priority as "do not recalculate", at least not without setting the prio to be directly less than zero which would lead to immediate update, so "update 7 days from now" is not *directly* achievable. If we could figure out a way to mark special priorities while keeping it a number, that would be best....

**Comment by ChristianBeer** [comment]
What about the obvious? negative numbers? Give them the same meaning (near to zero means more frequent update, far away less frequent) but make update_priorities ignore them. This would give them special meaning and we just need to find out what 7days from now means in a number.

**Comment by dubslow** [comment]
Well the problem with negative numbers is that the entire AllSeq.json file is organized as a minheap based on that priority, so all negative numbers will be automatically at the bottom of the minheap -- immediately updated. No real easy way to ignore them, because that would mean skipping a sizeable fraction of the minheap to get to the positive priorities, so then why bother with a minheap in the first place? Not to mention doing such would also render the priority column in the HTML table completely pointless, since sorting by priority would no longer bring up the next sequences to be queried.

**Comment by ChristianBeer** [comment]
You're right, I forgot about the minheap. Well it seems we need a new field in the file then?

**Comment by dubslow** [comment]
I'm really disinclined to add a new column just for this.

For now I'll implement the one-month-max-for-reserverds (when I get around to actually tackling all three current outstanding issues, should be this weekend hopefully), but I'll leave this issue open afterwards for further consideration.

**Comment by UweBeckert** [comment]
If yafu@home has finished a composite it is reported to factordb and a new composite from factordb is fetched. Yafu keeps a list of such updated sequences here https://yafu.myfirewall.org/yafu/download/ali/AliquotSeqences.html. This is also available as csv: https://yafu.myfirewall.org/yafu/download/ali/AliquotSeqences.
(thinkingg loud)
Maybe you can use this csv to update the blue page and only if the last update is too far away fetch the update from factordb.
